pkg/Vagrantfile

```diff
@@ -0,0 +1,56 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# Required modules
+require 'yaml'
+
+# Minimum Requirements
+Vagrant.require_version '>= 1.6.0'
+VAGRANTFILE_API_VERSION = '2'
+
+# Configuration files
+CONFIGURATION_FILE = 'config/bdb-config.yaml'
+
+# Validate if all the required plugins are present
+required_plugins = ["vagrant-cachier"]
+required_plugins.each do |plugin|
+  if not Vagrant.has_plugin?(plugin)
+    raise "Required vagrant plugin #{plugin} not found. Please run `vagrant plugin install #{plugin}`"
+  end
+end
+
+# Read configuration file(s)
+instances_config = YAML.load_file(File.join(File.dirname(__FILE__), CONFIGURATION_FILE))
+
+#TODO: (muawiakh) Add support for Docker, AWS, Azure
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  instances_config.each do |instance|
+    config.vm.define instance['name'] do |bdb|
+      # Workaround until vagrant cachier plugin supports dnf
+      if !(instance["box"]["name"].include? "fedora")
+        if Vagrant.has_plugin?("vagrant-cachier")
+          config.cache.scope = :box
+        end
+      end
+      bdb.vm.hostname = instance["name"]
+      bdb.vm.network instance["network"]["type"], ip: instance["network"]["ip"]
+      bdb.vm.box = instance["box"]["name"]
+      bdb.vm.synced_folder ".", "/bigchaindb"
+      bdb.vm.provision :shell, inline: "cd /bigchaindb/scripts;/bin/bash #{instance["upstart"]}"
+      if instance["setup_type"] == "quickstart"
+        bdb.vm.provision :shell, inline: "PYTHONBUFFERED=1 ansible-playbook \
+          /bigchaindb/ansible/quickstart.yml --extra-vars \"with_docker=#{instance["deploy_docker"]}\" -c local"
+      end
+
+      bdb.vm.provider 'vmware_fusion' do |vmwf, override|
+        vmwf.vmx['memsize'] = instance["ram"]
+        vmwf.vmx['numvcpus'] = instance['vcpus']
+      end
+
+      bdb.vm.provider 'virtualbox' do |vb, override|
+        vb.memory = instance["ram"]
+        vb.cpus = instance['vcpus']
+      end
+    end
+  end
+end
```

Comment:
Question: Can a used deploy a BDB vagrant box locally and then use the driver from the host to send transactions?
From what I understand looking at the networking, this only supports a private network which means that the user has to send transactions from within the VM. This can make user not use the local deployment for testing at all, as s/he will have to copy scripts to the box and back.
This can also hamper our plan to run a test setup using vagrant, as we will have to bundle all the tests in the vagrant box.

Can we change the networking to bridged or host-only mode to allow this to happen?
Or should we add this constraint to the deployment and do the testing in a different way?

Reply:
You can still run the tests/send transactions from the host, we don't need to copy them inside the `vagrant box`, I send transactions from my host by specifying the `bdb_root_url` of the bigchaindb process running inside the box.

Reply:
So, you mean vagrant takes care of resolving the BDB vagrant box name?

Reply:
On second thought, I have added support for `public network` as well, so this should not be a problem.